File: Chap1-Events/1-4EventPropagationDemo/parentbutton.cpp
#include "parentbutton.h"
#include <QDebug>

ParentButton::ParentButton(QWidget *parent) : QPushButton(parent)
{

}

void ParentButton::mousePressEvent(QMouseEvent *event)
{
    qDebug() << "ParentButton mousePressEvent Called";
    QPushButton::mousePressEvent(event);
}

File: Chap1-Events/1-4EventPropagationDemo/parentbutton.py
from PySide6.QtWidgets import QPushButton
from PySide6.QtGui import QMouseEvent

class ParentButton(QPushButton):
    def __init__(self, parent=None):
        super().__init__(parent)
    
    def mousePressEvent(self, event: QMouseEvent):
        print("ParentButton mousePressEvent Called")
        super().mousePressEvent(event)